### ISSUE TITLE
[indent] TypeError: Cannot read property 'range' of undefined

### ISSUE THREAD

**Issue by urielvan** [typescript-eslint/typescript-eslint#1115]
**Repro**

```JSON
{
  "rules": {
    "indent": "off",
    "@typescript-eslint/indent": [ "error", 2 ]
  }
}
```

```TS
/*
 * before entering `<>` after the keyword function, it works fine.
 * but with the empty parameter list, it throws error
 * btw, it is ok if entering `<T>`  in sequence.
 */
Array.prototype.foo = function<> (): void {
  console.log(1);
}

interface Array<T> {
  foo (): void;
}
```

**Expected Result**
no errors should be shown in eslint output

**Actual Result**
TypeError: Cannot read property 'range' of undefined
Occurred while linting <...>\eslint-test\index.ts:1
    at Object.JSXOpeningElement [as listener] (<...>\eslint-test\node_modules\eslint\lib\rules\indent.js:1449:53)
    at Program:exit.listenerCallQueue.filter.forEach.nodeInfo (<...>\eslint-test\node_modules\eslint\lib\rules\indent.js:1585:55)
    at Array.forEach (<anonymous>)
    at Program:exit (<...>\eslint-test\node_modules\eslint\lib\rules\indent.js:1585:26)
    at listeners.(anonymous function).forEach.listener (<...>\eslint-test\node_modules\eslint\lib\linter\safe-emitter.js:45:58)
    at Array.forEach (<anonymous>)
    at Object.emit (<...>\eslint-test\node_modules\eslint\lib\linter\safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (<...>\eslint-test\node_modules\eslint\lib\linter\node-event-generator.js:253:26)
    at NodeEventGenerator.applySelectors (<...>\eslint-test\node_modules\eslint\lib\linter\node-event-generator.js:282:22)
    at NodeEventGenerator.leaveNode (<...>\eslint-test\node_modules\eslint\lib\linter\node-event-generator.js:305:14)

**Additional Info**
Log taken with `./node_modules/.bin/eslint --ext .ts ./index.ts --debug &> eslint-log.txt`
[eslint-log.txt](https://github.com/typescript-eslint/typescript-eslint/files/3748792/eslint-log.txt)

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.4.0` |
| `@typescript-eslint/parser`        | `2.4.0` |
| `TypeScript`                       | `3.6.4` |
| `ESLint`                           | `6.5.1` |
| `node`                             | `12.12.0` |
| `npm`                              | `6.11.3` |
